src/assets/js/modules/todoList.js

```diff
@@ -0,0 +1,232 @@
+export const todoList = () => {
+  /**
+   * @type {HTMLElement}
+   */
+  const taskNameInputElement = document.querySelector(
+    '[data-input="taskname"]'
+  );
+  /**
+   * @type {HTMLElement}
+   */
+  const taskDeadlineElement = document.querySelector('[data-input="deadline"]');
+  /**
+   * @type {HTMLElement}
+   */
+  const taskAddButtonElement = document.querySelector(
+    '[data-input="addbutton"]'
+  );
+  /**
+   * @type {HTMLElement}
+   */
+  const taskListDoingElement = document.querySelector(
+    '[data-taskList="doing"]'
+  );
+  /**
+   * @type {HTMLElement}
+   */
+  const taskListDoneElement = document.querySelector('[data-taskList="done"]');
+
+  if (
+    !taskNameInputElement ||
+    !taskDeadlineElement ||
+    !taskAddButtonElement ||
+    !taskListDoingElement ||
+    !taskListDoneElement
+  )
+    return;
+
+  /**
+   * @type {NodeList}
+   */
+  const taskListItemDoingElements = taskListDoingElement.childNodes;
+
+  /**
+   * @type {NodeList}
+   */
+  const taskListItemDoneElements = taskListDoneElement.childNodes;
+
+  /**
+   * @type {object[]}
+   */
+  const taskArray = [];
+
+  /**
+   * 初期で表示されているタスクを進行中タスク配列もしくは完了後タスクに追加する関数
+   * @function
+   */
+  const addTaskObjToTaskArray = (nodeList) => {
+    nodeList.forEach((elm) => {
+      if (elm.nodeName === "LI") {
+        const isCompleted = elm.querySelector("input").checked;
+        const taskName = elm.querySelector("label").innerText;
+        const taskDeadline =
+          elm.querySelector("span").innerText !== ""
+            ? new Date(elm.querySelector("span").innerText)
+            : "none";
+        const taskObj = {
+          taskName,
+          taskDeadline,
+          isCompleted,
+        };
+        taskArray.push(taskObj);
+      }
+    });
+  };
+
+  /**
+   * タスク追加ボタンの活性非活性を切り替える関数
+   * @function
+   */
+  const switchAddButtonDisabled = () => {
+    const isInputLength = taskNameInputElement.value.length;
+    if (isInputLength) {
+      taskAddButtonElement.disabled = false;
+    } else {
+      taskAddButtonElement.disabled = true;
+    }
+  };
+
+  /**
+   * タスク入力input要素をリセットする関数
+   * @function
+   */
+  const resetTaskInputStatus = () => {
+    taskNameInputElement.value = "";
+    taskDeadlineElement.value = "";
+    taskAddButtonElement.disabled = true;
+  };
+
+  /**
+   * タスクオブジェクトを進行中タスク配列に追加する関数
+   * @function
+   */
+  const addTaskObjFromInputToTaskArray = () => {
+    const taskName = taskNameInputElement.value;
+    const taskDeadline = taskDeadlineElement.value
+      ? new Date(taskDeadlineElement.value)
+      : "none";
+    const taskObj = {
+      taskName,
+      taskDeadline,
+      isCompleted: false,
+    };
+    taskArray.push(taskObj);
+  };
+
+  /**
+   * 対応中タスク配列から対応済みタスク配列へタスクを移動させる関数
+   * @function
+   * @param {*} event
+   */
+  const switchTaskIsCompleted = (event) => {
+    const targetElement = event.target;
+    const targetElementNodeName = targetElement.nodeName;
```

Comment:
関数スコープ内であれば分割代入をよく使いますね

```suggestion
    const { target } = event;
    const { nodeName } = targetElement;
```

Reply:
`event.target`も分割代入使えるんですね…！！
修正しました！
7acb61dc9e217d862e86e9358b2afadb2b903e61

nodeNameの方は取得方法変更しました！
3ef9dab5b7fcd980e8188748dd29317916912c6a

Reply:
👍 いただいているのでresolveします！